### ISSUE TITLE
Support for soundtypes

### ISSUE THREAD

**Issue by rlnt** [Yesterday17/ZenScript#27]
Would be nice if soundtypes would be supported in the dump probe creates.

ContentTweaker has a feature that allows modpack makers to create custom blocks.
You can define a block sound type with a bracket handler. These bracket handlers are not yet dumped and therefor ZenScript shows an error that it's not available.

Example code:
```
var infusionBlock = VanillaFactory.createBlock("infusion_block", <blockmaterial:rock>);
infusionBlock.setBlockSoundType(<soundtype:stone>);
infusionBlock.register();
```

**Comment by Landiie** [comment]
been struggling with this too, really wish there was support for soundtype dumping. Is there anyway i could hide the errors for the soundtypes in the meantime? its pretty jarring D:

EDIT: nevermind! `#ignoreBracketErrors` works c:

hope this soundtype enhancement can be implemented soon! would be super cool

**Comment by Yesterday17** [comment]
A new strategy will be introduced to ZenScript in the next release to help the extension distinguish whether a bad bracketHandler is an unknown type.

What you need to do only is to disable `modIdItemCompletion` in options. This might be frustrating if you're used to using bracketHandler such as `<minecraft:white_wool>` to refer to white wool without the `item:` prefix(as it's optional). You may need to add all those prefixes manually.

The reason is quite simple: if you use `<modid:item_name>` to refer to items, it's impossible for the extension to know whether `modid` is an unknown type of `bracketHandler`, or just a mistyped normal bracketHandler.

In short, after disabling `modIdItemCompletion`, no error would be produced by unknown bracketHandlers.

**Comment by Landiie** [comment]
Ive disabled the `modIdItemCompletion` although i use your probe mod and i do quite enjoy the auto-fill features it has so i can do scripts without having to open the game sometimes, its really a time saver considering my pack's load times!

So ill just stick with putting `#ignoreBracketHandlers` on my script though until there is some workaround or like you said; will just flag the bracket as unknown rather than bad so there is no errors. Thanks for the input though!!
